helper/profiles/response_source.go

```diff
@@ -0,0 +1,94 @@
+package profiles
+
+import (
+	"context"
+	"fmt"
+)
+
+// ResponseSourceBuilder allows reading inputs from past responses.
+func ResponseSourceBuilder(ctx context.Context, engine *ProfileEngine, field map[string]interface{}) (Source, error) {
+	s := &ResponseSource{
+		outer: engine.outerBlockName,
+		field: field,
+	}
+
+	return s, nil
+}
+
+var _ SourceBuilder = ResponseSourceBuilder
+
+func WithResponseSource() func(*ProfileEngine) {
+	return func(p *ProfileEngine) {
+		p.sourceBuilders["response"] = ResponseSourceBuilder
+	}
+}
+
+type ResponseSource struct {
+	outer string
+	field map[string]interface{}
+
+	outerName     string
+	responseName  string
+	fieldSelector interface{}
+}
+
+var _ Source = &ResponseSource{}
+
+func (s *ResponseSource) Validate(_ context.Context) ([]string, []string, error) {
+	var responseName string
+
+	if s.outer != "" {
+		outerFieldName := fmt.Sprintf("%v_name", s.outer)
+		rawOuterName, present := s.field[outerFieldName]
+		if !present {
+			return nil, nil, fmt.Errorf("response source is missing required field '%v'", outerFieldName)
+		}
+
+		outerName, ok := rawOuterName.(string)
+		if !ok {
+			return nil, nil, fmt.Errorf("field '%v' is of wrong type: expected 'string' got '%T'", outerFieldName, rawOuterName)
```

Comment:
these are all nitpicks to use the `%q` formatting
```suggestion
			return nil, nil, fmt.Errorf("field %q is of wrong type: expected 'string' got '%T'", outerFieldName, rawOuterName)
```